File: src/app/services/team.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class TeamService {
  //Team : BE server address
  TeamURL: string = 'http://localhost:8086/api/teams';
  //httpclient : livreur / bostagi
  constructor(private httpClient: HttpClient) {}
  //array of objects
  getAllTeams() {
    return this.httpClient.get(this.TeamURL);
  }
  //object
  getTeamById(id: number) {
    // return this.httpClient.get(this.TeamURL+"/"+id)
    return this.httpClient.get(`${this.TeamURL}/${id}`);
  }
  
  //json Object {id:...,scoreOne:....,scoreTwo:....}
  addTeam(Team: any) {
    return this.httpClient.post(this.TeamURL, Team);
  }
  //json Object {id:...,scoreOne:....,scoreTwo:....}
  editTeam(TeamObj: any) {
    return this.httpClient.put(this.TeamURL, TeamObj);
  }
  //object, boolean , string
  deleteTeamById(id: number) {
    return this.httpClient.delete(`${this.TeamURL}/${id}`);
  }
  }

File: .vscode/springaout/src/main/java/com/example/webdevteam/springaout/services/TeamService.java
package com.example.webdevteam.springaout.services;

import java.util.List;
import com.example.webdevteam.springaout.models.TeamModel;

public interface TeamService {
    public List<TeamModel> getAllTeams();
    public TeamModel getTeamById(Long id);
    public void deleteTeamById(Long id);
    public TeamModel addTeam(TeamModel team);
    public TeamModel updateTeam(TeamModel team);
}
